packages/contracts-bedrock/scripts/checks/unused-imports/main.go

```diff
@@ -0,0 +1,149 @@
+package main
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strings"
+	"sync"
+	"sync/atomic"
+)
+
+var importPattern = regexp.MustCompile(`import\s*{([^}]+)}`)
+var asPattern = regexp.MustCompile(`(\S+)\s+as\s+(\S+)`)
+
+func main() {
+	if err := run(); err != nil {
+		writeStderr("an error occurred: %v", err)
+		os.Exit(1)
+	}
+}
+
+func writeStderr(msg string, args ...any) {
+	_, _ = fmt.Fprintf(os.Stderr, msg+"\n", args...)
+}
+
+func run() error {
+	cwd, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("failed to get current working directory: %w", err)
+	}
+
+	var hasErr int32
+	var outMtx sync.Mutex
```

Comment:
Suggesting this again: https://pkg.go.dev/golang.org/x/sync/errgroup

Reply:
Will make another PR that changes this for all of the different tasks at the same time